tutorials/stats-source-space/20_cluster_1samp_spatiotemporal.py

```diff
@@ -144,50 +144,75 @@
-# just pass the spatial adjacency matrix (instead of spatio-temporal)
+# For cluster-based permutation testing, we must define adjacency relations
+# that govern which points can become members of the same cluster. While
+# these relations are rather obvious for dimensions such as time or frequency
+# they require a bit more work for spatial dimension such as channels or
+# source vertices.
+#
+# Here, to use an algorithm optimized for spatio-temporal clustering, we
+# just pass the spatial adjacency matrix (instead of spatio-temporal).
+# But note that clustering still takes place along the
+# temporal dimension and can be
+# controlled via the ``max_step`` parameter in
+# :func:`mne.stats.spatio_temporal_cluster_1samp_test`.
+#
+# If we wanted to specify adjacencies for both space and time explicitly
+# we would have to use :func:`mne.stats.combine_adjacency`, however for
+# the present case, this is not needed.
 print('Computing adjacency.')
 adjacency = mne.spatial_src_adjacency(src)
 
-#    Note that X needs to be a multi-dimensional array of shape
-#    samples (subjects) x time x space, so we permute dimensions
+# %%
+# Compute statistic
+# -----------------
+
+# Note that X needs to be a multi-dimensional array of shape
+# observations (subjects) × time × space, so we permute dimensions
 X = np.transpose(X, [2, 1, 0])
 
-#    Now let's actually do the clustering. This can take a long time...
-#    Here we set the threshold quite high to reduce computation.
+# Here we set a cluster forming threshold based on a p-value for
+# the cluster based permutation test.
+# We use a two-tailed threshold, the "1 - p_threshold" is needed
+# because for two-tailed tests we must specify a positive threshold.
 p_threshold = 0.001
-t_threshold = -stats.distributions.t.ppf(p_threshold / 2., n_subjects - 1)
+df = n_subjects - 1  # degrees of freedom for the test
+t_threshold = stats.distributions.t.ppf(1 - p_threshold / 2, df=df)
+
+# Now let's actually do the clustering. This can take a long time...
 print('Clustering.')
 T_obs, clusters, cluster_p_values, H0 = clu = \
     spatio_temporal_cluster_1samp_test(X, adjacency=adjacency, n_jobs=None,
                                        threshold=t_threshold, buffer_size=None,
                                        verbose=True)
-#    Now select the clusters that are sig. at p < 0.05 (note that this value
-#    is multiple-comparisons corrected).
+
+# Now select the clusters that are statistically significant at p < 0.05
+# (note that this value is multiple-comparisons corrected).
 good_cluster_inds = np.where(cluster_p_values < 0.05)[0]
 
 # %%
 # Visualize the clusters
 # ----------------------
 print('Visualizing clusters.')
 
-#    Now let's build a convenient representation of each cluster, where each
-#    cluster becomes a "time point" in the SourceEstimate
+# Now let's build a convenient representation of each cluster, where each
+# cluster becomes a "time point" in the SourceEstimate
```

Comment:
Sure, here's my suggestion:
```suggestion
# Now let's build a convenient representation of our results, where consecutive
# cluster spatial maps are stacked in the time dimension of a SourceEstimate
# object. This way by moving through the time dimension we will be able to see
# subsequent cluster maps.
```
you can judge if this is an improvement or not :)